src/proc.cpp

```diff
@@ -674,7 +675,7 @@ static bool process_clean_after_marking(parser_t &parser, bool allow_interactive
             // don't create an event or it's easy to get an infinite loop.
             if (!j->from_event_handler() && j->should_report_process_exits()) {
                 pid_t pgid = *j->get_pgid();
-                exit_events.push_back(event_t::job_exit(pgid));
+                exit_events.push_back(event_t::job_exit(pgid, j->internal_job_id));
```

Comment:
I'm not entirely sure why this still uses the pgid? The method itself has it renamed to "pid", so wouldn't that be the last pid then, now? Or is it even necessary at all to store it in this case, if we match all pids anyway?

Reply:
It was historical, but you're right, nice catch. I switched this to passing the last pid instead of the process group.